src/biokbase/narrative/jobs/jobcomm.py

```diff
@@ -116,16 +175,24 @@ def __init__(self):
                 "start_job_update": self._modify_job_update,
                 "stop_job_update": self._modify_job_update,
                 "cancel_job": self._cancel_job,
-                "retry_job": self._retry_job,
+                "retry_job": self._retry_jobs,
                 "job_logs": self._get_job_logs,
                 "job_logs_latest": self._get_job_logs,
             }
 
     def _verify_job_id(self, req: JobRequest) -> None:
-        if req.job_id is None:
+        if not req.job_id:
             self.send_error_message("job_does_not_exist", req)
             raise ValueError(f"Job id required to process {req.request} request")
 
+    def _verify_job_id_list(self, req: JobRequest) -> None:
+        if not isinstance(req.job_id_list, list):
+            raise ValueError("List expected for job_id_list")
+        req.job_id_list[:] = [job_id for job_id in req.job_id_list if job_id]
```

Comment:
I think the original list data structure is also referenced in `req.rq_data`. It's never used, but I thought it would be more correct to change the contents of the underlying list rather than just assign a new list